flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/data/RowDataProjection.java

```diff
@@ -134,11 +176,26 @@ private static RowData.FieldGetter createFieldGetter(RowType rowType,
 
   public RowData wrap(RowData row) {
     this.rowData = row;
+    this.valueCache.clear();
     return this;
   }
 
   private Object getValue(int pos) {
-    return getters[pos].getFieldOrNull(rowData);
+    return valueCache.computeIfAbsent(pos, key -> {
+      Object value = null;
+      if (projectedFields != null) {
```

Comment:
Fetch data from iceberg nested struct data according to `projectedFields` and the field pos in flink output schema